clang/test/SemaSYCL/call-to-undefined-function.cpp

```diff
@@ -0,0 +1,183 @@
+// RUN: %clang_cc1 -fsycl-is-device -verify -fsyntax-only %s
+
+void defined() {
+  // empty
+}
+
+void undefined();
+// expected-note@-1 {{'undefined' declared here}}
+
+SYCL_EXTERNAL void undefinedExternal();
+
+template <typename name, typename Func>
+__attribute__((sycl_kernel)) void kernel_single_task(Func kernelFunc) {
+  kernelFunc();
+}
+
+template <typename T>
+void definedTpl() {
+  // empty
+}
+
+template <typename T>
+void undefinedTpl();
+// expected-note@-1 {{'undefinedTpl<int>' declared here}}
+
+template <typename T>
+extern SYCL_EXTERNAL void undefinedExternalTpl();
+
+template <typename T, bool X>
+void definedPartialTpl() {
+  // empty
+}
+
+template <>
+void definedPartialTpl<char, true>() {
+  // empty
+}
+
+template <typename T, bool X>
+struct Tpl {
+  void defined() {
+    // empty
+  }
+};
+
+template <typename T>
+struct Tpl<T, true> {
+  void defined() {
+    // empty
+  }
+};
+
+template <typename T, bool X>
+struct TplWithTplMethod {
+  template <typename T2, bool Y>
+  void defined() {
+    // empty
+  }
+};
+
+template <typename T>
+struct TplWithTplMethod<T, true> {
+  template <typename T2, bool Y>
+  void defined() {
+    // empty
+  }
+};
+
+template <typename T, bool X>
+struct TplWithTplMethod2 {
+  template <typename T2, bool Y>
+  void defined() {
+    // empty
```

Comment:
I suggest removing all `// empty` comments. They are useless and just make it harder to read the code.